tests/tests/tier0/bluechi-agent-user-bus/test_bluechi_agent_user_bus.py

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+import time
+from typing import Dict
+
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.service import Option, Section
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+    service = "org.eclipse.bluechi.Agent"
+    object = "/org/eclipse/bluechi"
+    interface = "org.eclipse.bluechi.Agent"
+    bluechi_user = "bluechiuser"
+    node_foo = nodes[NODE_FOO]
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    ctrl.systemctl.stop_unit("bluechi-controller")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+
+    bc_agent = node_foo.load_systemd_service(
+        directory="/usr/lib/systemd/system", name="bluechi-agent.service"
+    )
+    exec_start = bc_agent.get_option(Section.Service, Option.ExecStart)
+    bc_agent.set_option(Section.Service, Option.ExecStart, exec_start + " -u")
+    bc_agent.set_option(Section.Service, Option.User, "bluechiuser")
+    node_foo.install_systemd_service(bc_agent, restart=False)
+
+    node_foo.exec_run(f"useradd {bluechi_user}")
+    # Start the service as user ${bluechi_user}, start sleep for 5 seconds and detach
+    # The sleep&detach needed, because the moment the command is finished - user bus is closed
+    node_foo.exec_run(
+        f"runuser -l {bluechi_user} -c 'systemctl --user start bluechi-agent && sleep 5 &'"
+    )
+    time.sleep(1)
+    # Check that bluechi-agent connected to user bus
+    _, output = node_foo.exec_run(
+        f"busctl --machine={bluechi_user}@.host --user get-property {service} {object} {interface} Status"
+    )
+    assert output == 's "offline"'
```

Comment:
How about creating a python/bash script listening for the name acquired signal like in this gist: (using `UserMessageBus` instead of the system one, of course)
https://gist.github.com/engelmi/8bc6a02bbb721d2232d0db48c7cefcfa
The script should block and contain a timeout. 

And then replace the `sleep 5`, e.g.:
```suggestion
    result, output = node_foo.exec_run(
        f"runuser -l {bluechi_user} -c 'systemctl --user start bluechi-agent && python wait.py &'"
    )
    
    # assuming we return exit code 0 when successful
    assert result == 0
```

Reply:
Changed `run_python` to be able to run as user and dumped the logic there.
Implemented with polling, listening on `SessionMessageBus` for some reason didn't give the expected results